images/Dockerfile

```diff
@@ -1,36 +1,34 @@
+ARG DOCKER_VERSION=27.1.1-cli
+ARG RUNNER_CONTAINER_HOOKS_VERSION=0.6.1
+
+# Create a versioned alias for docker:cli
+FROM docker:${DOCKER_VERSION} AS docker-cli
+
 # Source: https://github.com/dotnet/dotnet-docker
-FROM mcr.microsoft.com/dotnet/runtime-deps:6.0-jammy as build
+FROM mcr.microsoft.com/dotnet/runtime-deps:6.0-jammy AS build
 
 ARG TARGETOS
 ARG TARGETARCH
 ARG RUNNER_VERSION
-ARG RUNNER_CONTAINER_HOOKS_VERSION=0.6.1
-ARG DOCKER_VERSION=27.1.1
-ARG BUILDX_VERSION=0.16.2
-
-RUN apt update -y && apt install curl unzip -y
+ARG RUNNER_CONTAINER_HOOKS_VERSION
 
 WORKDIR /actions-runner
-RUN export RUNNER_ARCH=${TARGETARCH} \
-    && if [ "$RUNNER_ARCH" = "amd64" ]; then export RUNNER_ARCH=x64 ; fi \
-    && curl -f -L -o runner.tar.gz https://github.com/actions/runner/releases/download/v${RUNNER_VERSION}/actions-runner-${TARGETOS}-${RUNNER_ARCH}-${RUNNER_VERSION}.tar.gz \
-    && tar xzf ./runner.tar.gz \
-    && rm runner.tar.gz
 
-RUN curl -f -L -o runner-container-hooks.zip https://github.com/actions/runner-container-hooks/releases/download/v${RUNNER_CONTAINER_HOOKS_VERSION}/actions-runner-hooks-k8s-${RUNNER_CONTAINER_HOOKS_VERSION}.zip \
-    && unzip ./runner-container-hooks.zip -d ./k8s \
-    && rm runner-container-hooks.zip
+# Configure git-core/ppa based on guidance here: https://git-scm.com/download/linux
+# The second `apt-get update` isn't strictly necessary, but let's leave it here to force the
+# image build to break if the ppa is suddenly unavailable.
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends curl unzip sudo lsb-release gpg-agent software-properties-common \
```

Comment:
Could you maybe add the comment

`# 'gpg-agent' and 'software-properties-common' are needed for the 'add-apt-repository' command that follows`

here so it is clear, why you need this?

But why the others? Is `sudo` and `lsb-release` already needed at this stage?